src/simd/distances_avx512.cc

```diff
@@ -36,6 +38,37 @@ masked_read(int d, const float* x) {
     return _mm_load_ps(buf);
     // cannot use AVX2 _mm_mask_set1_epi32
 }
+namespace {
+inline __m512
+avx512_bf16_patch(__m512&& f512) {
+    if (!enable_avx512_patch_fp32_bf16) {
```

Comment:
Will this runtime branching hurts performance? Is it necessary to template it?

Reply:
branching before call the dist cmp function, rather than within this function?

Reply:
Branch prediction should hit all targets, so the overhead can be ignored